src/main/java/UI/LoginUI.java

```diff
@@ -37,7 +37,7 @@ public LoginUI(WindowManager existingWindows, LoginController controller) {
         this.configureFrame();
         this.centreWindow();
         this.setListeners();
-        this.setVisible(true);
+        this.programWindows.openWindow(WindowManager.LOGIN_REFERENCE_KEY);
```

Comment:
This is definitely a good convention to have; that way any opening and closing of windows is always delegated to our WindowManager.